pkg/sharings/send_mails.go

```diff
@@ -5,206 +5,106 @@ import (
 	"net/url"
 
 	"github.com/cozy/cozy-stack/pkg/consts"
-	"github.com/cozy/cozy-stack/pkg/contacts"
 	"github.com/cozy/cozy-stack/pkg/couchdb"
 	"github.com/cozy/cozy-stack/pkg/globals"
 	"github.com/cozy/cozy-stack/pkg/instance"
 	"github.com/cozy/cozy-stack/pkg/jobs"
 	"github.com/cozy/cozy-stack/pkg/workers/mails"
 )
 
-// The sharing-dependant information: the recipient's name, the sharer's public
-// name, the description of the sharing, and the OAuth query string.
-type mailTemplateValues struct {
+// MailTemplateValues is a struct with the recipient's name, the sharer's
+// public name, the description of the sharing, and the link to the sharing.
+type MailTemplateValues struct {
 	RecipientName    string
 	SharerPublicName string
 	Description      string
 	SharingLink      string
 }
 
-// SendDiscoveryMail send a mail to the recipient, in order for him to give his
-// URL to the sender
-func SendDiscoveryMail(instance *instance.Instance, s *Sharing, rs *RecipientStatus) error {
+// SendMails send an email to each recipient, so that they can accept or
+// refuse the sharing.
+func SendMails(instance *instance.Instance, s *Sharing) error {
 	sharerPublicName, err := instance.PublicName()
-	if err != nil {
-		return err
+	if err != nil || sharerPublicName == "" {
+		sharerPublicName = "Someone"
 	}
 	// Fill in the description.
 	desc := s.Description
 	if desc == "" {
-		desc = "[No description provided]"
-	}
-	discoveryLink, err := generateDiscoveryLink(instance, s, rs)
-	if err != nil {
-		return err
-	}
-
-	// Generate the base values of the email to send
-	discoveryMsg, err := generateMailMessage(s, rs.recipient,
-		&mailTemplateValues{
-			RecipientName:    rs.recipient.Email[0].Address,
-			SharerPublicName: sharerPublicName,
-			Description:      desc,
-			SharingLink:      discoveryLink,
-		},
-	)
-	if err != nil {
-		return err
-	}
-	_, err = globals.GetBroker().PushJob(&jobs.JobRequest{
-		Domain:     instance.Domain,
-		WorkerType: "sendmail",
-		Options:    nil,
-		Message:    discoveryMsg,
-	})
-	return err
-}
-
-// SendSharingMails will generate the mail containing the details
-// regarding this sharing, and will then send it to all the recipients.
-func SendSharingMails(instance *instance.Instance, s *Sharing) error {
-	sharerPublicName, err := instance.PublicName()
-	if err != nil {
-		return err
-	}
-	// Fill in the description.
-	desc := s.Description
-	if desc == "" {
-		desc = "[No description provided]"
+		desc = "Surprise!"
```

Comment:
nit: you could use `i18n.Translate` to internationalize this pseudo-description.